src/I18n/Parser/PoFileParser.php

```diff
@@ -142,11 +142,10 @@ protected function _addMessage(array &$messages, array $item)
 
         $translation = stripcslashes($translation);
 
-        if ($context && (!isset($messages[$singular]) || is_array($messages[$singular]))) {
+        if ($context !== null) {
             $messages[$singular]['_context'][$context] = $translation;
-        }
-        if ($context === null) {
-            $messages[$singular] = $translation;
+        } else {
+            $messages[$singular]['_context'][''] = $translation;
```

Comment:
Do you think the additional overhead is justified here in terms of functional completeness/expectation? IMO those improvements here trump the overhead - at least the silent dropping of translations sounds rather like a hard to debug and understandable issue otherwise.
Or is there any better way of dealing with it that would make this less lengthening?